src/core/ipfs/config.js

```diff
@@ -1,14 +1,56 @@
 'use strict'
 
+const promisify = require('promisify-es6')
+const _get = require('lodash.get')
+const _set = require('lodash.set')
+
 module.exports = function config (self) {
   return {
-    // cli only feature built with show and replace
-    // edit: (callback) => {},
-    replace: (config, callback) => {
+    get: promisify((key, callback) => {
+      if (typeof key === 'function') {
+        callback = key
+        key = undefined
+      }
+
+      if (!key) {
+        return self._repo.config.get(callback)
+      }
+
+      if (typeof key !== 'string') {
+        return callback(new Error('Invalid key type'))
+      }
+
+      self._repo.config.get((err, config) => {
+        if (err) {
+          return callback(err)
+        }
+        const value = _get(config, key, undefined)
+        if (!value) {
+          callback(new Error('Key does not exist in config'))
```

Comment:
This is not entirely accurate, `value` could be an empty string for `0`, and would as such evaluate to `true` in this check.